logstash-plugins/logstash-input-google_pubsub
# Update deps and modernize gradle

## Release notes

Updates PubSub dependencies forcing this plugin to be compatible only with Logstash `>= 8.2.0`

## What does this PR do?

- Updates Gradle from 4.0 to 7.4.2
- Update Google Guava from 20.0 to 31.1 (fix issue #57)
- Update Google Protobuf from 3.6.0 to 3.21.4
-  Removes `7.x` branch testing from Travis matrix test


## Why is it important/What is the impact to the user?
Keeps the build system and dependencies updated

## Checklist




## Author's Checklist



## How to test this PR locally

- Checkout this PR locally
- Unpack a local Logstash distribution tar.gz (`>=8.2.0`) 
- Build the gem
  -  In shell export
  ```sh
  export LOGSTASH_PATH=/path/to/logstash/logstash/ && export LOGSTASH_SOURCE=1
  ```
  - vendor the gem:
  ```sh
  ./gradlew clean vendor
  ```
- Install the gem into Logstash, update the `gemfile.lock` with:
```
gem "logstash-input-google_pubsub", :path => "/path/to/the/repo//logstash-input-google_pubsub"
```
- install the plugin code
```
bin/logstash-plugin install --no-verify
```
- run a pipeline which uses this plugin
- verify it's fine


## Related issues


- Closes #57

## Use cases



## Screenshots



## Logs
